### PR TITLE
Fix Bean trying to be registered more then one time

### LINKED ISSUE #108: Application cannot register SSM client
**Type**: Bug

**Component**:
Parameter Store

**Describe the bug**
Currently we try to migrate our services to Spring Boot 2.4.4, Spring Cloud 2020.0.2 and Spring Cloud AWS 2.3.
As far as I understand the bootstrap context has been deprecated by Spring Cloud Commons and the new way to include configuration via AWS Parameter Store would be
```
spring:
  config:
    import: "aws-parameterstore:"
```
which we have in `application-aws.yml`.

Now after deployment we currently end up with
```
java.lang.IllegalStateException: Could not register object [com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClient@785a4557] under bean name 'configDataAWSSimpleSystemsManagement': there is already object [com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClient@785a4557] bound
    at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.registerSingleton(DefaultSingletonBeanRegistry.java:124)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.registerSingleton(DefaultListableBeanFactory.java:1138)
    at io.awspring.cloud.autoconfigure.paramstore.AwsParamStoreConfigDataLocationResolver.lambda$registerAndPromoteBean$1(AwsParamStoreConfigDataLocationResolver.java:106)
    at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176)
    at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:169)
    at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143)
    at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:131)
    at org.springframework.boot.DefaultBootstrapContext.close(DefaultBootstrapContext.java:133)
    at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:392)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:325)
    at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:144)
    at org.springframework.cloud.stream.binder.DefaultBinderFactory.getBinderInstance(DefaultBinderFactory.java:338)
    at org.springframework.cloud.stream.binder.DefaultBinderFactory.doGetBinder(DefaultBinderFactory.java:221)
    at org.springframework.cloud.stream.binder.DefaultBinderFactory.getBinder(DefaultBinderFactory.java:149)
    at org.springframework.cloud.stream.function.FunctionConfiguration$FunctionToDestinationBinder.bindFunctionToDestinations(FunctionConfiguration.java:443)
    at org.springframework.cloud.stream.function.FunctionConfiguration$FunctionToDestinationBinder.afterPropertiesSet(FunctionConfiguration.java:401)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524)
    at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
    at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
    at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
    at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:944)
    at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
    at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
    at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:63)
    at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:769)
    at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:761)
    at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:426)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:326)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1313)
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302)
    at my.ApplicationKt.main(Application.kt:48)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.base/java.lang.reflect.Method.invoke(Unknown Source)
    at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
    at org.springframework.boot.loader.Launcher.launch(Launcher.java:107)
    at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
    at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
```

Any suggestions what's going on here?

### PR DESCRIPTION
**Pull request by MatejNedic**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Fixes #108 
Fixes same bug but for Secret Manager load as well

## :pencil: Checklist

- [x] I reviewed submitted code
- [] I added tests to verify changes
- [x] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Review by maciejwalkowiak — COMMENTED** [review]
Looks good! Thanks @MatejNedic for fixing. Ideally, please add test to make sure that it does not happen again.

File: spring-cloud-starter-aws-parameter-store-config/src/main/java/io/awspring/cloud/autoconfigure/paramstore/AwsParamStoreConfigDataLocationResolver.java [hunk 1]
```diff
@@ -103,8 +104,11 @@ public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDat
 		registerBean(context, type, supplier);
 		context.getBootstrapContext().addCloseListener(event -> {
 			T instance = event.getBootstrapContext().get(type);
-			event.getApplicationContext().getBeanFactory().registerSingleton("configData" + type.getSimpleName(),
-					instance);
+			String name = "configData" + type.getSimpleName();
+			ConfigurableApplicationContext appContext = event.getApplicationContext();
```
Comment (maciejwalkowiak) [inline spring-cloud-starter-aws-parameter-store-config/src/main/java/io/awspring/cloud/autoconfigure/paramstore/AwsParamStoreConfigDataLocationResolver.java:108]: Please add a comment here and in corresponding Secrets Manager class explaining why this check is necessary. It won't take too much time until it is forgotten.

**Review by maciejwalkowiak — APPROVED** [review]

**Review by eddumelendez — APPROVED** [review]

**Comment by AppSecConcierge** [comment]
When is this going to be released into Maven?

**Comment by maciejwalkowiak** [comment]
If all goes with a plan next week

**Pull request merged**